pkgs/top-level/variants.nix

```diff
@@ -95,6 +95,26 @@ self: super: {
     };
   });
 
+  # This maps each entry in _cuda.bootstrapData.cudaCapabilityToInfo to its own package set. Each of these will contain
+  # all packages built for that target system.
+  # For instance, pkgsForCudaArch.sm_90a.python3Packages.torch will refer to PyTorch built for the Hopper architecture,
+  # leveraging architecture-specific features.
+  # NOTE: Not every package set is supported on every architecture!
+  # See `Using pkgsForCudaArch` in doc/languages-frameworks/cuda.section.md for more information.
+  pkgsForCudaArch = lib.mapAttrs' (cudaCapability: cudaCapabilityInfo: {
+    name = self._cuda.lib.mkRealArchitecture cudaCapability;
+    value = nixpkgsFun {
+      config = super.config // {
+        cudaSupport = true;
+        rocmSupport = false;
+        # Not supported by architecture-specific feature sets, so disable for all.
+        # Users can choose to build for family-specific feature sets if they wish.
+        cudaForwardCompat = false;
+        cudaCapabilities = [ cudaCapabilityInfo.cudaCapability ];
+      };
+    };
+  }) self._cuda.bootstrapData.cudaCapabilityToInfo;
```

Comment:
1. Why `bootstrapData` rather than the merged `db`?
2. If we use the merged `db`, do we increase the top-level pkgs' eval cost?

Reply:
1. Agreed, switched to `db`.
2. We won't know until the DB PR is merged ;)